### PR TITLE
feat: add GET /connect/session endpoint

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes

depends on https://github.com/NangoHQ/nango/pull/2769

- add auth middleware to be able to authenticate via the connect session token
- add GET /connect/session endpoint
- add DELETE /connect/session endpoint

## Issue ticket number and link

https://linear.app/nango/issue/NAN-1775/[server]-api-endpoints-to-getpost-sessions

### CODE REVIEW

File: packages/server/lib/controllers/connect/deleteSession.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    const emptyBody = requireEmptyBody(req);
+    if (emptyBody) {
+        res.status(400).send({ error: { code: 'invalid_body', errors: zodErrorToHTTP(emptyBody.error) } });
+        return;
+    }
+
+    const deleteSession = await connectSessionService.deleteConnectSession(db.knex, {
+        id: res.locals.connectSession.id,
+        accountId: res.locals.account.id,
+        environmentId: res.locals.environment.id
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/connect/deleteSession.ts]: using res.locals everywhere makes things a bit less clear imo

File: packages/server/lib/middleware/access.middleware.ts [hunk 2]
```diff
@@ -167,6 +169,58 @@ export class AccessMiddleware {
...
+
+            const token = authorizationHeader.split('Bearer ').pop();
+
+            if (!token) {
+                return errorManager.errRes(res, 'malformed_auth_header');
+            }
+
+            const getConnectSession = await connectSessionService.getConnectSessionByToken(db.knex, token);
+            if (getConnectSession.isErr()) {
+                return errorManager.errRes(res, 'unknown_connect_token');
```
Comment (bodinsamuel) [inline packages/server/lib/middleware/access.middleware.ts]: needs to be added to default ApiError response

File: packages/server/lib/services/connectSession.service.ts [hunk 3]
```diff
@@ -104,6 +105,19 @@ export async function getConnectSession(
...
+    const getSession = await keystore.getPrivateKey(db, token);
+    if (getSession.isErr()) {
+        return Err(new ConnectSessionError({ code: 'not_found', message: `Token not found`, payload: { token: `${token.substring(0, 32)}...` } }));
+    }
+    const privateKey = getSession.value;
+    const session = await getConnectSession(db, { id: privateKey.entityId, accountId: privateKey.accountId, environmentId: privateKey.environmentId });
+    if (session.isErr()) {
+        return Err(session.error);
+    }
+    return Ok(session.value);
```
Comment (bodinsamuel) [inline packages/server/lib/services/connectSession.service.ts:118]: you can't return session directly?

File: packages/types/lib/connect/api.ts [hunk 4]
```diff
@@ -19,8 +20,33 @@ export type PostConnectSessions = Endpoint<{
         allowedIntegrations?: string[] | undefined;
         integrationsConfigDefaults?: Record<string, { connectionConfig: Record<string, unknown> }> | undefined;
     };
-    Error: ApiError<'forbidden' | 'invalid_body' | 'invalid_query_params' | 'internal_error'>;
+    Error: ApiError<'forbidden'>;
```
Comment (bodinsamuel) [inline packages/types/lib/connect/api.ts]: doesn't seem to be returned anyway

File: packages/types/lib/connect/api.ts [hunk 5]
```diff
@@ -19,8 +20,33 @@ export type PostConnectSessions = Endpoint<{
...
     Success: {
         data: ConnectSessionToken;
     };
 }>;
+
+export type GetConnectSession = Endpoint<{
+    Method: 'GET';
+    Path: '/connect/session';
+    Error: ApiError<'forbidden'>;
```
Comment (bodinsamuel) [inline packages/types/lib/connect/api.ts]: same

**Review by bodinsamuel — APPROVED** [review]

File: packages/server/lib/services/connectSession.service.ts (hunk 3 above)
Comment (TBonnin) [reply packages/server/lib/services/connectSession.service.ts:118]: why not?

Comment (bodinsamuel) [reply packages/server/lib/services/connectSession.service.ts:118]: I meant this: 
```suggestion
    return session
```

instead of the current if err ok

Comment (TBonnin) [reply packages/server/lib/services/connectSession.service.ts:118]: ok yes. you are right. the unwrapping/wrapping is unnecessary. I will fix next time I touch the code

**Pull request merged**